src/services/validationService.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  ICommonRules,
+  IRules,
+  IValidationResponse,
+  INativeQuery,
+  IDataSourceRules,
+  IQueryTypeRules,
+  IQuery,
+  ISqlQuery,
+} from "../models";
+import constants from "../resources/constants.json";
+import { isEmpty, isUndefined } from "lodash";
+import moment from "moment";
+import fs from "fs";
+import Ajv from "ajv";
+
+export class ValidationService {
+  public static validateNativeQuery(
+    query: INativeQuery,
+    commonLimits: ICommonRules,
+    dataSourceLimits: IDataSourceRules
+  ): IValidationResponse {
+    let nativeQuerySchema = JSON.parse(
+      fs.readFileSync(
+        process.cwd() + "/src/configs/nativeQuerySchema.json",
+        "utf8"
+      )
+    );
+    let ajv = new Ajv();
+    let isValidQuery = ajv.validate(
+      nativeQuerySchema,
+      JSON.parse(JSON.stringify(query))
+    );
+    if (isValidQuery) {
+      if (!isEmpty(dataSourceLimits) && !isUndefined(dataSourceLimits)) {
+        this.determineQueryLimits(query, commonLimits);
+        try {
+          return this.validateQueryRules(
+            query,
+            dataSourceLimits.queryRules[
+              query.queryType as keyof IQueryTypeRules
+            ]
+          );
+        } catch (error) {
+          return { errorMessage: constants.INVALID_QUERY_TYPE, isValid: false };
+        }
+      } else {
+        return {
+          errorMessage: constants.INVALID_DATA_SOURCE,
+          isValid: false,
+        };
+      }
+    } else {
+      let error = ajv.errors;
+      let message =
+        error![0].instancePath.replace("/", "") + " " + error![0].message;
+      return { errorMessage: message, isValid: false };
+    }
+  }
+
+  private static validateQueryRules(
+    query: INativeQuery,
+    queryRules: IRules
+  ): IValidationResponse {
+    const dateRange: any = query.intervals;
+    let allowedRange = queryRules.max_date_range;
+    const extractedDateRange = Array.isArray(dateRange)
+      ? dateRange[0].split("/")
+      : dateRange.toString().split("/");
+    const fromDate = moment(extractedDateRange[0], "YYYY-MM-DD HH:MI:SS");
+    const toDate = moment(extractedDateRange[1], "YYYY-MM-DD HH:MI:SS");
+    return this.validateDateRange(fromDate, toDate, allowedRange);
+  }
+
+  private static determineQueryLimits(
+    query: INativeQuery,
+    limits: ICommonRules
```

Comment:
I don't think we required to update the query limits in the method. Just validate the query limit and return the validation status object.